File: examples/python/local_agent_pair.py
from __future__ import annotations

import os
import tempfile

from conu_sdk import ConuClient


def main() -> None:
    home = os.path.join(tempfile.gettempdir(), "conu-python-example")
    client = ConuClient(home=home)

    client.init()
    client.register_agent("agent.python.alice", "Alice Python Agent")
    client.register_agent("agent.python.bob", "Bob Python Agent")
    client.process_queued()

    result = client.send_message(
        "agent.python.alice",
        "agent.python.bob",
        b"example private payload",
    )
    client.process_queued()

    print(f"queued request {result['requestId']}")
    print(f"opaque bytes {result['payloadBytes']}")
    print("payload contents were not displayed by conU")


if __name__ == "__main__":
    main()

File: examples/typescript/local_agent_pair.mjs
import { ConuClient } from "../../sdk/typescript/src/index.js";

const client = new ConuClient({ home: ".conu-typescript-example" });

client.init();
client.registerAgent("agent.alpha", "Alpha", { streams: true, rooms: true });
client.registerAgent("agent.beta", "Beta", { streams: true, rooms: true });
client.processQueued();

const sent = client.sendMessage("agent.alpha", "agent.beta", Buffer.from("private bytes"));
client.processQueued();
const inbox = client.inbox("agent.beta");
const firstEnvelopeId = Array.isArray(inbox.entries) ? inbox.entries[0]?.envelopeId : undefined;
const received = firstEnvelopeId
  ? client.receiveMessageBytes("agent.beta", firstEnvelopeId)
  : Buffer.alloc(0);

console.log(
  JSON.stringify(
    {
      sentEnvelopeId: sent.envelopeId,
      inboxEntries: Array.isArray(inbox.entries) ? inbox.entries.length : 0,
      receivedBytes: received.byteLength,
      contentsDisplayed: false,
    },
    null,
    2,
  ),
);
